TESTS/netsocket/tls/tlssocket_echotest.cpp

```diff
@@ -102,29 +102,31 @@ void TLSSOCKET_ECHOTEST()
 
 void tlssocket_echotest_nonblock_receive()
 {
-    int recvd = sock->recv(&(tls_global::rx_buffer[bytes2recv_total - bytes2recv]), bytes2recv);
-    if (recvd == NSAPI_ERROR_WOULD_BLOCK) {
-        if (tc_exec_time.read() >= time_allotted) {
+    while (bytes2recv > 0) {
+        int recvd = sock->recv(&(tls_global::rx_buffer[bytes2recv_total - bytes2recv]), bytes2recv);
+        if (recvd == NSAPI_ERROR_WOULD_BLOCK) {
+            if (tc_exec_time.read() >= time_allotted) {
+                receive_error = true;
+            }
+            return;
+        } else if (recvd < 0) {
             receive_error = true;
+        } else {
+            bytes2recv -= recvd;
         }
-        return;
-    } else if (recvd < 0) {
-        receive_error = true;
-    } else {
-        bytes2recv -= recvd;
-    }
 
-    if (bytes2recv == 0) {
-        TEST_ASSERT_EQUAL(0, memcmp(tls_global::tx_buffer, tls_global::rx_buffer, bytes2recv_total));
+        if (bytes2recv == 0) {
+            TEST_ASSERT_EQUAL(0, memcmp(tls_global::tx_buffer, tls_global::rx_buffer, bytes2recv_total));
 
-        static int round = 0;
-        printf("[Recevr#%02d] bytes received: %d\n", round++, bytes2recv_total);
-        tx_sem.release();
-    } else if (receive_error || bytes2recv < 0) {
-        TEST_FAIL();
-        tx_sem.release();
+            static int round = 0;
```

Comment:
These debug prints could be removed as those don't bring any value anymore. Those were more helpful when writing the test cases.